File: modules/edu.gemini.seqexec.server/src/main/java/edu/gemini/seqexec/server/osgi/Activator.java
package edu.gemini.seqexec.server.osgi;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;
/**
 * Created by jluhrs on 4/17/15.
 */
public class Activator implements BundleActivator {
    @Override
    public void start(BundleContext bundleContext) throws Exception {
        System.out.println("Hello World!");
    }

    @Override
    public void stop(BundleContext bundleContext) throws Exception {

    }
}

File: modules/edu.gemini.seqexec.shared/src/main/scala/edu/gemini/seqexec/shared/osgi/Activator.scala
package edu.gemini.seqexec.shared.osgi

import java.util

import edu.gemini.pot.spdb.IDBDatabaseService
import edu.gemini.seqexec.shared.{SeqExecService, SeqExecServer}
import org.osgi.framework.{ServiceRegistration, BundleContext, BundleActivator}

import edu.gemini.util.osgi.Tracker._
import org.osgi.util.tracker.ServiceTracker


object Activator {
  val PublishTrpc = "trpc"
}

class Activator extends BundleActivator {
  private var tracker: Option[ServiceTracker[_,_]] = None

  override def start(ctx: BundleContext): Unit = {
    import Activator._

    tracker = Some(track[IDBDatabaseService, ServiceRegistration[_]](ctx) { (odb) =>
      val seqServer = new SeqExecServer(odb)

      val props = new util.Hashtable[String, Object]()
      props.put(PublishTrpc, "true")
      ctx.registerService(classOf[SeqExecService], seqServer, props)
    } { _.unregister() })

    tracker.foreach(_.open())
  }

  override  def stop(ctx: BundleContext): Unit = {
    tracker.foreach(_.close())
    tracker = None
  }
}
